pkg/build/builder/util.go

```diff
@@ -55,11 +58,65 @@ func getDockerNetworkMode() stiapi.DockerNetworkMode {
 	defer file.Close()
 
 	if id := readNetClsCGroup(file); id != "" {
-		return stiapi.NewDockerNetworkModeContainer(id)
+		return s2iapi.NewDockerNetworkModeContainer(id)
 	}
 	return ""
 }
 
+// GetCGroupLimits returns a struct populated with cgroup limit values gathered
+// from the local /sys/fs/cgroup filesystem.  Overflow values are set to
+// math.MaxInt64.
+func GetCGroupLimits() (*s2iapi.CGroupLimits, error) {
+	byteLimit, err := readInt64("/sys/fs/cgroup/memory/memory.limit_in_bytes")
+	if err != nil {
+		return nil, err
+	}
+
+	cpuQuota, err := readInt64("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us")
+	if err != nil {
+		return nil, err
+	}
+
+	cpuShares, err := readInt64("/sys/fs/cgroup/cpu,cpuacct/cpu.shares")
+	if err != nil {
+		return nil, err
+	}
+
+	cpuPeriod, err := readInt64("/sys/fs/cgroup/cpu,cpuacct/cpu.cfs_period_us")
+	if err != nil {
+		return nil, err
+	}
+	return &s2iapi.CGroupLimits{
+		CPUShares:        cpuShares,
+		CPUPeriod:        cpuPeriod,
+		CPUQuota:         cpuQuota,
+		MemoryLimitBytes: byteLimit,
+		// Set memoryswap==memorylimit, this ensures no swapping occurs.
+		// see: https://docs.docker.com/engine/reference/run/#runtime-constraints-on-cpu-and-memory
+		MemorySwap: byteLimit,
+	}, nil
+}
+
+func readInt64(filePath string) (int64, error) {
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		return -1, err
+	}
+	s := strings.TrimSpace(string(data))
+	val, err := strconv.ParseInt(s, 10, 64)
+	// overflow errors are ok, we'll get return a math.MaxInt64 value which is more
+	// than enough anyway.  For underflow we'll return MinInt64 and the error.
+	if err != nil && err.(*strconv.NumError).Err == strconv.ErrRange {
+		if s[0] == '-' {
+			return math.MinInt64, err
+		}
+		return math.MaxInt64, nil
```

Comment:
I still think that returning 0 in case of under/overflow is clearer.

I've used 0 and observed that the subcontainer has the same memory constraints as the parent (18446744073709551615 was the number in bytes).

Returning MaxInt64 will be fine (it's an incredible amount of memory anyway) but makes the subcontainer have an unnecessarily (questionable?) different cgroup limits.

What's funny, using MaxInt64 (9223372036854775807), and printing `/sys/fs/cgroup/memory/memory.limit_in_bytes` from the subcontainer gives:
9223372036854775808